### PR TITLE
docs: System-wide profiling related updates

### PR DESCRIPTION
**Pull request by javierhonduco**
Signed-off-by: Francisco Javier Honduvilla Coto <javierhonduco@gmail.com>

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -6,35 +6,27 @@ Parca Agent uses BPF CO-RE (Compile Once – Run Everywhere) using [libbpf](http
 
 From a high-level it performs the following steps:
 
-* [Target Discovery](#target-discovery): Discovers cgroups to attach the profiler on the machine the Parca Agent is running on.
+* [Target Discovery](#target-discovery): Discovers Kubernetes clusters, if any, and fetches its metadata.
```
Comment (kakkoyun) [inline docs/design.md:9]: I think this should be called something else.
Our targets now are all the processes on the system.

We should add a step to describe how we provide context for the processes using container runtimes, cgroups and systems units.

Comment (javierhonduco) [reply docs/design.md:9]: "Metadata Discovery"?

**Comment by javierhonduco** [comment]
This PR updates all the docs, except `design.md` which I will update in another PR (cutting an issue for that one)

File: docs/design.md (hunk 1 above)
Comment (kakkoyun) [reply docs/design.md]: Both sound great. You pick one :)

**Pull request merged**